sdk/eventhub/Azure.Messaging.EventHubs/stress/src/Infrastructure/Publisher.cs

```diff
@@ -0,0 +1,174 @@
+                                sendTasks.Add(Task.Run(async () =>
+                                {
+                                    while (!backgroundCancellationSource.Token.IsCancellationRequested)
+                                    {
+                                        await PerformSend(producer, backgroundCancellationSource.Token).ConfigureAwait(false);
+
+                                        if ((testConfiguration.ProducerPublishingDelay.HasValue) && (testConfiguration.ProducerPublishingDelay.Value > TimeSpan.Zero))
+                                        {
+                                            await Task.Delay(testConfiguration.ProducerPublishingDelay.Value, backgroundCancellationSource.Token).ConfigureAwait(false);
+                                        }
+                                    }
+                                }));
+                            }
+                        }
+
+                        // Perform one of the sends in the foreground, which will allow easier detection of a
+                        // processor-level issue.
+
+                        while (!cancellationToken.IsCancellationRequested)
+                        {
+                            try
+                            {
+                                await PerformSend(producer, cancellationToken).ConfigureAwait(false);
+
+                                if ((testConfiguration.ProducerPublishingDelay.HasValue) && (testConfiguration.ProducerPublishingDelay.Value > TimeSpan.Zero))
+                                {
+                                    await Task.Delay(testConfiguration.ProducerPublishingDelay.Value, cancellationToken).ConfigureAwait(false);
+                                }
+                            }
+                            catch (TaskCanceledException)
+                            {
+                                backgroundCancellationSource.Cancel();
+                                await Task.WhenAll(sendTasks).ConfigureAwait(false);
+                            }
+                        }
+                    }
+                }
+                catch (TaskCanceledException)
+                {
+                    // No action needed.
+                }
+                catch (Exception ex) when
+                    (ex is OutOfMemoryException
+                    || ex is StackOverflowException
+                    || ex is ThreadAbortException)
+                {
+                    throw;
+                }
+                catch (Exception ex)
+                {
+                    // If this catch is hit, there's a problem with the producer itself; cancel the
+                    // background sending and wait before allowing the producer to restart.
+
+                    backgroundCancellationSource.Cancel();
+                    await Task.WhenAll(sendTasks).ConfigureAwait(false);
+
+                    //Interlocked.Increment(ref Metrics.ProducerRestarted);
```

Comment:
We'll definitely want an equivalent for this.  It is used to tell us if the producer is crashing, and we had to restart it.